File: client-leger/app/src/main/java/com/auth0/androidlogin/models/IMatch.kt
package com.auth0.androidlogin.models

import classes.Team
import com.google.gson.annotations.SerializedName

open class IMatch(
    @SerializedName("game")
    val game: IGame,

    @SerializedName("begin")
    val begin: String,

    @SerializedName("end")
    val end: String,

    @SerializedName("bestScore")
    val bestScore: Int,

    @SerializedName("accessCode")
    val accessCode: String,

    @SerializedName("testing")
    val testing: Boolean,

    @SerializedName("players")
    var players:  MutableList<IPlayer>,

    @SerializedName("observers")
    var observers:  MutableList<IObserver>,

    @SerializedName("managerName")
    val managerName: String,

    @SerializedName("managerId")
    val managerId: String,

    @SerializedName("isAccessible")
    var isAccessible: Boolean,

    @SerializedName("isFriendMatch")
    var isFriendMatch: Boolean,

    @SerializedName("bannedNames")
    val bannedNames: MutableList<String>,

    @SerializedName("playerAnswers")
    var playerAnswers: MutableList<IPlayerAnswers>,

    @SerializedName("panicMode")
    val panicMode: Boolean,

    @SerializedName("timer")
    val timer: Int,

    @SerializedName("timing")
    val timing: Boolean,

    @SerializedName("isTeamMatch")
    var isTeamMatch: Boolean,

    @SerializedName("isPricedMatch")
    var isPricedMatch: Boolean,

    @SerializedName("priceMatch")
    var priceMatch: Number = 0,

    @SerializedName("nbPlayersJoined")
    var nbPlayersJoined: Number = 0,

    @SerializedName("teams")
    var teams: MutableList<Team> = mutableListOf(),

    @SerializedName("currentQuestionIndex")
    val currentQuestionIndex: Number
)

File: server/app/interfaces/i-match.ts
import { PlayerAnswers } from '@app/classes/player-answers/player-answers';
import { Game } from '@app/model/database/game';
import { Player } from './player';

/**
 * Interface of all attribute of a game mostly use the create a game
 */
export interface IMatch {
    game: Game;
    begin: string;
    end: string;
    bestScore: number;
    accessCode: string;
    testing: boolean;
    players: Player[];
    managerName: string;
    managerId: string;
    isAccessible: boolean;
    isFriendMatch: boolean;
    bannedNames: string[];
    playerAnswers: PlayerAnswers[];
    panicMode: boolean;
    timer: number;
    timing: boolean;
    isTeamMatch: boolean;
    isPricedMatch: boolean;
    priceMatch: number;
    nbPlayersJoined: number;
}
